src/docfx/SubCommands/MetadataMerger.cs

```diff
@@ -104,6 +111,103 @@ private static void MergeToc(MetadataMergeParameters parameters, string outputBa
                 YamlMime.TableOfContent);
         }
 
+        private void InitMetaTable(FileModel model)
+        {
+            var content = model.Content as PageViewModel;
+            if (content != null && model.Uids.Length > 0)
+            {
+                _metaTable.Add(model.Uids.First().Name, content.Metadata);
+            }
+        }
+
+        private void SetTocViewModelMetadata(TocViewModel vm, ImmutableList<string> metaNames)
+        {
+            foreach (var item in vm)
+            {
+                foreach (var metaName in metaNames)
+                {
+                    MergeIntoTocMetadata(item, metaName);
+                }
+            }
+        }
+
+        private List<string> MergeIntoTocMetadata(TocItemViewModel item, string metaName)
+        {
+            Dictionary<string, object> metadata;
+            if (_metaTable.TryGetValue(item.Uid, out metadata))
+            {
+                object metaValue;
+                if (metadata.TryGetValue(metaName, out metaValue))
+                {
+                    var merged = TryGetListFromObject(metaValue);
+                    foreach (var child in item.Items ?? Enumerable.Empty<TocItemViewModel>())
+                    {
+                        var childMetaValue = MergeIntoTocMetadata(child, metaName);
+                        merged = MergeMetadata(merged, childMetaValue);
```

Comment:
>merged = MergeMetadata(merged, childMetaValue); [](start = 24, length = 47)

Question: why the expand of metadata is from children to parent? namely, why children' metadata could be merged to the parent, not the opposite way?